common/regression_test.py

```diff
@@ -0,0 +1,26 @@
+import magma as m
+
+
+def check_interfaces(magma_circuit, genesis_circuit):
+    """
+    This checks that the interface to the genesis circuit is
+    compatible with the magma circuit. It does that by looping
+    over all the ports in the genesis circuit and checking that
+    the magma circuit has the same port and type.
+
+    It currently does not check the other direction (e.g. the magma circuit
+    could have ports not included in the genesis circuit interface)
+    """
+    genesis_port_names = genesis_circuit.interface.ports.keys()
+    for name in genesis_port_names:
+        assert hasattr(magma_circuit, name), \
+            f"Magma circuit does not have port {name}"
+        genesis_kind = type(type(getattr(genesis_circuit, name)))
+        magma_kind = type(type(getattr(magma_circuit, name)))
+        # Special case genesis signals that aren't typed
+        if name == "clk":
```

Comment:
these may not be 100% consistent across the board. we may want to make the clock and reset name parameters that default to "clk" and "reset"

Reply:
made the name->type mapping a parameter.